### PR TITLE
EVG-14646 Add json option to evergreen host list

### PR DESCRIPTION
**Pull request by Chukwurado**
[EVG-14646](https://jira.mongodb.org/browse/EVG-14646)

### Description 
Add option to print out list of hosts in json format.
Screen Shot 2022-01-13 at 3 45 33 PM

### CODE REVIEW

**Comment by ablack12** [comment]
@Chukwurado can you please share a screenshot of what this output looks like from staging or local testing?

**Comment by Chukwurado** [comment]
> @Chukwurado can you please share a screenshot of what this output looks like from staging or local testing?

@ablack12 just added in the description above

File: operations/host_spawn.go [hunk 1]
```diff
@@ -1018,6 +1028,29 @@ func printHosts(hosts []*restModel.APIHost) {
 	}
 }
 
+func printHostsJSON(hosts []*restModel.APIHost) {
+	var output strings.Builder
+	output.WriteString("[")
+	for i, h := range hosts {
```
Comment (hadjri) [inline operations/host_spawn.go:1034]: Can we try to simply marshal the host list to JSON and print it like the way we do here? https://github.com/evergreen-ci/evergreen/blob/main/operations/patch_list.go#L88 This seems cleaner than hardcoding.

**Review by hadjri — CHANGES_REQUESTED** [review]

File: operations/host_spawn.go [hunk 2]
```diff
@@ -1018,6 +1029,35 @@ func printHosts(hosts []*restModel.APIHost) {
...
 
+func printHostsJSON(hosts []*restModel.APIHost) {
+	type hostResult struct {
+		Id               string `json:"id"`
+		Name             string `json:"name"`
+		Distro           string `json:"distro"`
+		Status           string `json:"status"`
+		HostName         string `json:"hostName"`
+		User             string `json:"user"`
+		AvailabilityZone string `json:"availabilityZone"`
```
Comment (ablack12) [inline operations/host_spawn.go:1040]: We use snake case by convention for json, so can these be host_name and availability_zone?

**Comment by ablack12** [comment]
Also need to increment the ClientVersion string

**Review by ablack12 — APPROVED** [review]

**Review by hadjri — APPROVED** [review]

**Pull request merged**